### ISSUE TITLE
Integration Rational PoSt construction

### ISSUE THREAD

**Issue by anorth** [filecoin-project/go-filecoin#3307]
### Description
The rust-fil-proofs team are close to finishing implementation of the Rational PoSt construction. We wish to integrate this as soon as possible so as to include it in the long-running network release scheduled for late September.

For more background, see https://github.com/filecoin-project/rust-fil-proofs/issues/849

This issue is scoped specifically to work that needs to happen before the network is started. Work that can be deferred for a protocol upgrade of the running 0.5 network is less urgent, but we need to prepare now to make such work possible.

### Acceptance criteria
- go-filecoin uses the forthcoming Rational PoSt construction

### Risks + pitfalls
- proofs implementation is not yet complete, yet we want to start integration ASAP
- The rust-fil-proofs FFI may change
- Many PoSt characteristics may change dramatically (e.g. PoSt computation time), requiring protocol parameter changes to accomodate
